pandas/tests/series/test_arithmetic.py

```diff
@@ -169,3 +169,161 @@ def test_ser_cmp_result_names(self, names, op):
             ser = Series(cidx).rename(names[1])
             result = op(ser, cidx)
             assert result.name == names[2]
+
+# ------------------------------------------------------------------
+# Unary
+
+
+class TestSeriesUnary:
+
+    def test_neg(self):
+        ser = tm.makeStringSeries()
+        ser.name = 'series'
+        tm.assert_series_equal(-ser, -1 * ser)
+
+    def test_pos(self):
+        ser = tm.makeStringSeries()
+        ser.name = 'series'
+        tm.assert_series_equal(+ser, ser)
+
+    def test_invert(self):
+        ser = tm.makeStringSeries()
+        ser.name = 'series'
+        tm.assert_series_equal(-(ser < 0), ~(ser < 0))
+
+    @pytest.mark.parametrize('op', [operator.pos, operator.neg,
+                                    operator.inv, operator.abs])
+    def test_integer(self, any_numpy_ea_int_dtype, op):
+        # GH#23087
+        typ = any_numpy_ea_int_dtype
+
+        ser = Series([1, 3, 2], index=range(3), dtype=typ)
+        result = op(ser)
+        exp = Series(op(ser.values), index=ser.index)
+        tm.assert_series_equal(result, exp)
+
+    @pytest.mark.parametrize('op', [operator.pos, operator.neg,
+                                    operator.abs])
+    def test_float(self, float_dtype, op):
+        # GH#23087
+        ser = Series([1.1, 3.1, 2.1], index=range(3), dtype=float_dtype)
+        result = op(ser)
+        exp = Series(op(ser.values), index=ser.index)
+        tm.assert_series_equal(result, exp)
+
+    @pytest.mark.parametrize('op', [operator.inv])
+    def test_float_inv_raise(self, float_dtype, op):
+        # GH#23087
+        ser = Series([1.1, 3.1, 2.1], index=range(3), dtype=float_dtype)
+        with pytest.raises(TypeError):
+            op(ser)
+        # consistent with numpy
+        with pytest.raises(TypeError):
+            op(ser.values)
+
+    @pytest.mark.parametrize('typ', ['bool'])
+    @pytest.mark.parametrize('op', [operator.pos, operator.neg,
+                                    operator.inv, operator.abs])
+    def test_bool(self, typ, op):
+        # GH#23087
+        ser = Series([True, False, True], index=range(3), dtype=typ)
+
+        if op is operator.neg:
+            result = op(ser)
+            exp = Series(operator.inv(ser.values), index=ser.index)
+            tm.assert_series_equal(result, exp)
+
+            # inconsistent with NumPy
+            with pytest.raises(TypeError):
+                op(ser.values)
+        else:
+            result = op(ser)
+            if op is operator.pos and pd.compat.numpy._is_numpy_dev:
```

Comment:
Not sure how this is going to hold up. w.r.t the linked issue:

https://github.com/numpy/numpy/pull/11450

That was milestoned for numpy 1.16 but interestingly only appears in dev with this change.